### PR TITLE
feat(fn): implement new fn-shell component

### PR DESCRIPTION
**Pull request by mikerodonnell89**
Implements the new fn-shell component.
Screen Shot 2021-12-23 at 4 13 31 PM
Screen Shot 2021-12-23 at 4 13 27 PM
Screen Shot 2021-12-23 at 4 13 23 PM
Screen Shot 2021-12-23 at 4 13 18 PM

### CODE REVIEW

File: src/fn/fn-shell.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+.#{$block} {
+  @include fn-reset();
+
+  $fd-shell-action-buttons-padding: 0.4375rem;
+
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:14]: These 4 lines are a mixin `fn-flex-center `

File: src/fn/fn-shell.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  padding: 0.75rem 1.5rem;
+  background: #ffffff;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:16]: White color is a variable: `$fn-color-white`

File: src/fn/fn-shell.scss [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    @include fn-reset();
+
+    flex-grow: 1;
+    flex-shrink: 1;
+    flex-basis: 0;
+
+    &--product {
+      align-items: center;
+      display: flex;
+      justify-content: flex-start;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:29]:
```
@include fn-flex-vertical-center() {
    justify-content: flex-start;
}
```

File: src/fn/fn-shell.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      }
+
+      button {
+        @include fn-set-margins-x(-0.625rem, 0.75rem);
+      }
+
+      button, .#{$block}__logo {
+        align-items: center;
+        display: flex;
+        justify-content: center;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:54]: `@include fn-flex-center()`

File: src/fn/fn-shell.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    &--search {
+      justify-content: center;
+      max-width: 16rem;
+      order: 2;
+    }
+
+    &--actions {
+      align-items: center;
+      display: flex;
+      justify-content: flex-end;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:67]:
```
@include fn-flex-vertical-center() {
    justify-content: flex-end;
}
```

File: src/fn/fn-shell.scss [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    flex-basis: 0;
+
+    &--product {
+      align-items: center;
+      display: flex;
+      justify-content: flex-start;
+      order: 1;
+
+      .#{$block}__logo {
+        height: 1.625rem;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:33]: missing reset

File: src/fn/fn-shell.scss [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        width: 3.25rem;
+        vertical-align: middle;
+      }
+
+      .#{$block}__title {
+        @include fn-set-margin-left(0.375rem);
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:39]: missing reset

File: src/fn/fn-shell.scss [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+    flex-grow: 1;
+    flex-shrink: 1;
+    flex-basis: 0;
+
+    &--product {
+      align-items: center;
+      display: flex;
+      justify-content: flex-start;
+      order: 1;
```
Comment (InnaAtanasova) [inline src/fn/fn-shell.scss:30]: Why do you need order?

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/fn/fn-shell.scss (hunk 8 above)
Comment (mikerodonnell89) [reply src/fn/fn-shell.scss:30]: To lay out these elements in the correct order L>R. I guess this can be removed and we can leave it up to the developer

Comment (mikerodonnell89) [reply src/fn/fn-shell.scss]: This was to encourage the dev to place these elements in the correct spot in the shellbar, but yes that might be too opinionated, i'll remove this

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**